pool/entities.go

```diff
@@ -0,0 +1,170 @@
+package pool
+
+type ID string
+
+type Name string
+
+// I don't know i have to bring Peer Available storage and other metric inside Peer
+type Peer string
+
+type ReplicationFactor int8
+
+// Requirement of joining the pool
+type Requirement struct {
+	Latency   string `json:"latency"`
+	Bandwidth string `json:"bandwidth"`
+	ReplicationFactor ReplicationFactor `json:"rep_factor"`
+}
+
+// ResourceRequest object (it is much more complicated for simplicity we think as file).
+type Resource struct {
+	ID  ID `json:"id"`
+	FileName  Name
+	Owner     Peer
+	Size      string
+	ReplicationFactor ReplicationFactor
+	Allocation []Peer
+}
+
+
+// Event is a domain event marker.
+type Event interface {
+	isEvent()
+}
+
+func (e PoolCreated) isEvent() {}
+func (e ResourceRequested) isEvent() {}
+func (e ResourceAllocated) isEvent() {}
+func (e MemberJoined) isEvent() {}
+func (e MemberLeaved) isEvent() {}
+
+// PoolCreated Event
+type PoolCreated struct {
+	ID	         ID           `json:"id"`
+	Name         Name         `json:"name"`
+	Creator      Peer         `json:"creator"`
+	Members      []Peer       `json:"new_creator"`
+	Requirement  Requirement  `json:"new_requirement"`
+}
+
+// ResourceRequested Event
+type ResourceRequested struct {
+	ID               ID               `json:"id"`
+	ResourceRequest  Resource         `json:"new_resource"`
+}
+
+// ResourceAllocated Event
+type ResourceAllocated struct {
+	ID          ID          `json:"id"`
+	Resource  ID  `json:"new_resource"`
+	Allocation Peer
+}
+
+// MemberJoined Event
+type MemberJoined struct {
+	ID      ID    `json:"id"`
+	Member  Peer  `json:"member"`
+}
+
+// MemberLeaved Event
+type MemberLeaved struct {
+	ID      ID    `json:"id"`
+	Member  Peer  `json:"member"`
+}
+
+
+// Pool Aggregate
+type Pool struct {
+	id ID
+	name Name
+	creator Peer
+	members []Peer
+	requirement Requirement
+	resources []Resource
+
+	changes []Event
```

Comment:
This grows indefinitely? Does this cause any performance issues?

Reply:
:)) that's ok.